test/celeritas/geo/HeuristicGeoLauncher.hh

```diff
@@ -130,7 +134,7 @@ CELER_FUNCTION void HeuristicGeoLauncher::operator()(ThreadId tid) const
         geo.set_dir(dir);
```

Comment:
@sethrj  I noticed that the last argument to `rotate()` is constant, so direction is not actually changed currently, as the returned direction is discarded.

Reply:
Oh goodness, excellent catch @mrguilima ! Maybe we should rename the method `rotated`, and definitely add the [nodiscard](https://en.cppreference.com/w/cpp/language/attributes/nodiscard) attribute to the function when C++17 is required (or maybe add it now with a suitable macro).

Reply:
Woohoo, now I'm getting the same assertion we see in the failing testem3 + field for orange! (https://github.com/celeritas-project/celeritas/issues/460)

Reply:
It's as I feared; a direction change on the boundary that causes a track to be "reentrant" (originally outward direction leaving the volume, but then heads back in) causes the particle to enter the wrong volume, and then on the subsequent step it dies.

Reply:
OK, I'm going to leave this enabled now for vecgeom, call this MR a "vecgeom bugfix and identify orange bug", and create a separate MR to fix the reentrant track error.